Integrations/Flashpoint/Flashpoint.py

```diff
@@ -0,0 +1,1321 @@
+            "Flashpoint.Report.platform_url": report.get('platform_url', ''),
+            "Flashpoint.Report.title": report.get('title', ''),
+            "Flashpoint.Report.summary": report.get('summary', '')
+        }
+
+    else:
+        hr += 'No report found for the given ID.'
+        ec = {}
+
+    return hr, ec, resp
+
+
+def get_related_reports_command(client, report_id):
+    """
+    Get reports related to given report
+
+    :param report_id:
+    :param client:
+    :return: command output
+    """
+    url_suffix = '/reports/' + urllib.parse.quote(report_id.encode('utf-8')) + '/related?limit=5'
+    resp = client.http_request("GET", url_suffix=url_suffix)
+    reports = resp.get("data", [])
+
+    hr = '### Flashpoint Intelligence related reports:\n'
+
+    if reports:
+        hr += 'Top 5 related reports:\n\n'
+        report_details = []
+        index = 0
+        for report in reports:
+            title = report.get('title', 'N/A')
+            platform_url = report.get('platform_url', '')
+            summary = report.get('summary', 'N/A')
+            index += 1
+            hr += '' + str(index) + ') [{}]({})'.format(title, platform_url) + '\n'
+            hr += '   Summary: ' + str(summary) + '\n\n\n'
+            report_detail = {
+                'updated_at': report.get('updated_at', ''),
+                'posted_at': report.get('posted_at', ''),
+                'notified_at': report.get('notified_at', ''),
+                'platform_url': platform_url,
+                'title': title,
+                'summary': summary
+            }
+            report_details.append(report_detail)
+
+        fp_url = client.url + '/home/intelligence/reports/report/' + report_id + '#detail'
+        hr += 'Link to the given Report on Flashpoint platform: [{}]({})\n'.format(fp_url, fp_url)
+        ec: Dict[Any, Any] = {
+            "Flashpoint.Reports": report_details
+        }
+
+    else:
+        hr += 'No related reports found for the search.'
+        ec = {}
+
+    return hr, ec, resp
+
+
+def get_event_by_id_command(client, event_id):
+    """
+    Get specific event using its event id
+
+    :param client:
+    :param event_id: event's fpid
+    :return: command output
+    """
+    url_suffix = '/indicators/event/' + urllib.parse.quote(event_id.encode('utf-8'))
+    resp = client.http_request("GET", url_suffix=url_suffix)
+
+    hr = '### Flashpoint Event details\n'
+    ec: Dict[Any, Any] = {}
+
+    if len(resp) <= 0:
+        hr += 'No event found for the given ID.'
+        return hr, ec, resp
+
+    event = resp[0].get('Event', '')
+    fpid = resp[0].get('fpid', '')
+
+    events = []
+    if event:
+        event = convert_event(client, event, fpid)
+        events.append(event)
+        hr += tableToMarkdown('Below are the detail found:', events, ['Observed time (UTC)', 'Name', 'Tags'])
+
+        ec['Flashpoint'] = {
+            'event': {
```

Comment:
```suggestion
            'event(val.event_id == obj.event_id)': {
```
(val.event_id == obj.event_id)' this means aggregate by unique identifier event_id.
so if you get twice the same event, the demisto will aggregate those 2 events as single event according to event_id.
make sure you use the same functionality in every other place in the code where you have unique identifier fields for different objects.

Reply:
updated code to use the condition everywhere.